tests/projects/test_utils.py

```diff
@@ -92,6 +96,22 @@ def test__fetch_project(local_git_repo, local_git_repo_uri, zipped_repo, httpser
     assert os.path.exists(fetched_git_project)
 
 
+@pytest.mark.parametrize(
+    "version,expected_version,raises",
+    [
+        (None, "master", does_not_raise()),
+        (GIT_PROJECT_BRANCH, GIT_PROJECT_BRANCH, does_not_raise()),
+        ("non-version", "", pytest.raises(ExecutionException)),
+    ],
+)
+def test__fetch_git_repo(local_git_repo, local_git_repo_uri, version, expected_version, raises):
+    # Verify that the correct branch is checked out
+    with raises:
+        _fetch_git_repo(local_git_repo_uri, version, local_git_repo)
+        repo = git.Repo(local_git_repo)
+        assert repo.active_branch.name == expected_version
```

Comment:
Can we split this test like below?

```python
@pytest.mark.parametrize(
    "version,expected_version", [(None, "master"), (GIT_PROJECT_BRANCH, GIT_PROJECT_BRANCH)],
)
def test__fetch_git_repo(local_git_repo, local_git_repo_uri, version, expected_version):
    # Verify that the correct branch is checked out
    _fetch_git_repo(local_git_repo_uri, version, local_git_repo)
    repo = git.Repo(local_git_repo)
    assert repo.active_branch.name == expected_version


def test_fetching_non_existing_version_fails(local_git_repo, local_git_repo_uri):
    with pytest.raises(ExecutionException, match="Unable to checkout"):
        _fetch_git_repo(local_git_repo_uri, "non-version", local_git_repo)
```

Reply:
Hey @harupy, thanks for the pointer. The test is split up now